operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/TemporaryResourceCache.java

```diff
@@ -41,8 +42,15 @@ public TemporaryResourceCache(ManagedInformerEventSource<T, ?, ?> managedInforme
     this.managedInformerEventSource = managedInformerEventSource;
   }
 
-  public synchronized Optional<T> removeResourceFromCache(T resource) {
-    return Optional.ofNullable(cache.remove(ResourceID.fromResource(resource)));
+  public synchronized void onEvent(T resource, boolean unknownState) {
+    ResourceID resourceId = ResourceID.fromResource(resource);
+
+    T cachedResource = cache.get(resourceId);
+
+    if (unknownState || (cachedResource != null
+        && !isLaterResourceVersion(resourceId, cachedResource, resource))) {
```

Comment:
Kinda makes this code hard to read that the feature flag is checked inside the `isLaterResourceVersion`, would extract on a level above or rename this function to denote that fact.

Reply:
Will add a javadoc